text/0033-view-hierarchy.md

```diff
@@ -0,0 +1,116 @@
+```json
+{
+    "rendering_system": "compose", // or android_view_system, apple_uikit, apple_swiftui, unity, flutter, ...
+    "windows": [
+        {
+            "type": "com.example.ui.FancyButton",
+            "identifier": "R.id.button_logout",
+            "children": [], // nested elements
+            "width": 100.0,
+            "height": 100.0,
+            "depth": 100.0, // if applies
+            "x": 0.0,
+            "y": 0.0,
+            "z": 2.0, // if applies.
+            "visible": true|false,
+            "alpha": 1, // a float number from 0 to 1, where 0 means totally transparent, and 1 totally opaque.
+            "{extra_properties}": "{property value}" // additional platform-specific attributes
+        }
+    ]
+}
+```
+
+Some remarks on the example:
+ * `windows`: contains all visible windows, on mobile it's typically just one or two (e.g. if a dialog is open)
+ * `type`: The fully qualified widget class name, this name may be obfuscated on certain platforms (e.g. Android release builds with proguard enabled)
+ * `children` nests all child UI widgets, which then builds up the whole UI tree
+
+A typical Android/iOS view hierarchy for a single window consists of around 100-200 objects. Taking the attributes from the example above this generates a raw JSON file with a size of around 50KB. More complex view hierarchies for e.g. a Unity game may hold 1000-2000 objects, producing a JSON file of around 500KB.
+
+## (De)obfuscation
+As the view hierarchy may contain obfuscated attribute values (see `type` attribute in the example above) we need to process the attachment server-side, similar to what we already do for minidumps, [or profiles](https://github.com/getsentry/sentry/blob/cf71af372677487d7d0a7fd8ac9dd092f9596cf4/src/sentry/profiles/task.py#L350-L360).
+
+# Options Considered
+- Use a new envelope type instead of attachements: Whilst doable it's consideribly more work without adding any benefits. Using attachements has several advantages
+    - Attachments are billable by default
+    - We can immediately start sending and using this data, no need for large Relay updates
+    - Less new code in the SDKs when compared to adding a new envelope item, so less hassle to maintain/test
+    - We don't have to touch outcomes/rate limiting/processing code at all
+    - We always can eventually migrate to a new envelope item type if we feel the need to, that option is still left open.
+- The view hierarchy could be part of the Event, but because of the size limit, it is not recommended
+
+
+# Drawbacks
+- By utilizing attachments, [attachment scrubbing](https://docs.sentry.io/product/data-management-settings/scrubbing/attachment-scrubbing/) is relevant for any PII attributes in the view hierarchy
+- Increases event payload size
+- We've decided to not add the content of textfields, editables or labels because of PII.
+    - We may evaluate an options to allow PII.
+
+
+## Initial Proposal
```

Comment:
Will the initial proposal remain a part of the final doc? It's a bit confusing to read.

Reply:
I think we can remove it.